kolibri/core/assets/test/vue.download-button.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-env mocha */
+// The following two rules are disabled so that we can use anonymous functions with mocha
+// This allows the test instance to be properly referenced with `this`
+/* eslint prefer-arrow-callback: "off", func-names: "off" */
+
+'use strict';
+
+const Vue = require('vue-test');
+const downloadButton = require('../src/vue/content-renderer/download-button.vue');
+const DownloadButtonComponent = Vue.extend(downloadButton);
+const assert = require('assert');
+const simulant = require('simulant');
+
+describe('download-button Component', function () {
+  const samplesFiles = [{
+    file_size: 100000,
+    preset: 'High Resolution',
+    download_url: '/downloadcontent/3893fd801427402ad07487c5d2d35119.mp4/Fractions_High_Resolution.mp4',
+  }, {
+    file_size: 500,
+    preset: 'Thumbnail',
+    download_url: '/downloadcontent/187598e1f4596bf4492f5a205922b633.jpg/Fractions_Thumbnail.jpg',
+  }];
+
+
+  describe('computed property', function () {
+    describe('dropDownItems', function () {
+      beforeEach(function () {
+        this.vm = new DownloadButtonComponent({
+          propsData: {
+            files: samplesFiles,
+          },
+        }).$mount();
+      });
+      it('should return false if menu has never been opened', function () {
+        assert.equal(this.vm.dropdownOpenText, 'false');
+      });
+      it('should return true after opening menu', function () {
+        this.vm.toggleDropdown();
```

Comment:
Can possibly just directly set the state on the vm here, rather than relying on a method to do it, and then test the method separately for the job it should be doing.